lib/valkyrie/persistence/solr/repository.rb

```diff
@@ -39,20 +45,31 @@ def add_documents(documents)
     end
     # rubocop:enable Style/IfUnlessModifier
 
+    # Deletes a Solr Document using the ID
+    # @return [Array<Valkyrie::Resource>] resources which have been deleted from Solr
     def delete
       connection.delete_by_id resources.map { |resource| resource.id.to_s }, params: COMMIT_PARAMS
       resources
     end
 
+    # Given a Valkyrie Resource, generate the Hash for the Solr Document
+    # @param [Valkyrie::Resource] resource
+    # @return [Hash]
     def solr_document(resource)
       resource_factory.from_resource(resource: resource).to_h
     end
 
+    # Given a new Valkyrie Resource, generate a random UUID and assign it to the Resource
+    # @param [Valkyrie::Resource] resource
+    # @param [String] the UUID for the new resource
     def generate_id(resource)
       Valkyrie.logger.warn "The Solr adapter is not meant to persist new resources, but is now generating an ID."
       resource.id = SecureRandom.uuid
     end
 
+    # If a 409 error response is encountered when attempting to commit updates to Solr, raise a StaleObjectError
```

Comment:
Can you change `409` to `409 Conflict`? I think it's generally good form to include at least some part of the labe of the error codes for anyone who doesn't have them all memorized.  In fact, I would be 👍 on renaming this method to `handle_409_conflict` or even `handle_conflict`.